src/registry/tests.rs

```diff
@@ -0,0 +1,111 @@
+//! Tests for the registry
+
+use std::collections::BTreeMap;
+use std::sync::mpsc;
+use std::thread;
+
+use super::*;
+
+use rustc_serialize::{Decodable, Decoder};
+use rustc_serialize::json;
+use rustc_serialize::json::{Json, ToJson};
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+struct Point {
+    x: i32,
+    y: i32
+}
+
+impl ToJson for Point {
+    fn to_json(&self) -> Json {
+        let mut map = BTreeMap::new();
+        map.insert("x".to_string(), Json::I64(self.x as i64));
+        map.insert("y".to_string(), Json::I64(self.y as i64));
+        Json::Object(map)
```

Comment:
One thing I noticed is `BTreeMap` keeps things sorted. If you had two `HashMaps` with an "x" and "y" key, especially if they used different `Hasher`s, you couldn't just iterate through the keys in order and check when you're comparing two Json objects.